packages/devtools_app/lib/src/memory/flutter/memory_heatmap.dart

```diff
@@ -240,19 +285,50 @@ class FlameChartRenderObject extends RenderBox {
     markNeedsPaint();
   }
 
+  List<String> _exposeMatchNames(String searchName) {
+    return matchNodeNames(_sizes.root.children, searchName);
+  }
+
+  /// Look for the node with a particular name (depth first traversal).
+  List<String> matchNodeNames(Map<String, Node> children, String searchName) {
```

Comment:
does Node extends TreeNode? If so, consider adding a `depthFirstTraversal` method to trees.dart (similar to `breadthFirstTraversal`), and then you could add a param `depthFirst` to `firstChildWithCondition` and search by the appropriate traversal type based on the value of `depthFirst`. If you need multiple matches, create a method `childrenWithCondition` in trees.dart. 

It would be nice to keep tree code in that file, as it is well tested and easy to add more tests for.

Reply:
No, these are nodes in the heat map.